x-pack/plugins/ml/public/application/components/custom_urls/custom_url_editor/editor.tsx

```diff
@@ -360,66 +374,82 @@ export const CustomUrlEditor: FC<CustomUrlEditorProps> = ({
-            customTimeRange={customUrl?.customTimeRange}
-          />
-        ) : null}
 
-        {(type === URL_TYPE.KIBANA_DASHBOARD || type === URL_TYPE.KIBANA_DISCOVER) &&
-          showTimeRangeSelector && (
-            <>
-              <EuiSpacer size="m" />
-              <EuiFlexGroup>
-                <EuiFlexItem grow={false}>
-                  <EuiFormRow
-                    label={
-                      <FormattedMessage
-                        id="xpack.ml.customUrlsEditor.timeRangeLabel"
-                        defaultMessage="Time range"
-                      />
-                    }
-                    className="url-time-range"
-                    display="rowCompressed"
-                  >
-                    <EuiSelect
-                      options={timeRangeOptions}
-                      value={timeRange.type}
-                      onChange={onTimeRangeTypeChange}
-                      data-test-subj="mlJobCustomUrlTimeRangeInput"
-                      compressed
-                    />
-                  </EuiFormRow>
-                </EuiFlexItem>
-                {timeRange.type === TIME_RANGE_TYPE.INTERVAL && (
-                  <EuiFlexItem>
+        {(type === URL_TYPE.KIBANA_DASHBOARD || type === URL_TYPE.KIBANA_DISCOVER) && (
+          <>
+            <EuiSpacer size="m" />
+            <EuiFlexGroup direction="column" gutterSize="xs">
+              <EuiFlexItem grow={false}>
+                <EuiFlexGroup alignItems="center">
+                  <EuiFlexItem grow={false}>
                     <EuiFormRow
                       label={
                         <FormattedMessage
-                          id="xpack.ml.customUrlsEditor.intervalLabel"
-                          defaultMessage="Interval"
+                          id="xpack.ml.customUrlsEditor.timeRangeLabel"
+                          defaultMessage="Time range"
                         />
                       }
                       className="url-time-range"
-                      error={invalidIntervalError}
-                      isInvalid={isInvalidTimeRange}
                       display="rowCompressed"
                     >
-                      <EuiFieldText
-                        value={timeRange.interval}
-                        onChange={onTimeRangeIntervalChange}
-                        isInvalid={isInvalidTimeRange}
-                        data-test-subj="mlJobCustomUrlTimeRangeIntervalInput"
-                        compressed
-                      />
+                      {showCustomTimeRangeSelector && hasTimefield ? (
+                        <IntervalTimerangeSelector
+                          disabled={customUrl?.customTimeRange !== undefined}
+                          setAddIntervalTimerange={setAddIntervalTimerange}
+                          addIntervalTimerange={addIntervalTimerange}
+                        />
+                      ) : (
+                        <EuiSelect
```

Comment:
If an invalid interval is entered, the alignment is off here.

<img width="539" alt="image" src="https://github.com/elastic/kibana/assets/7405507/c79fa9a7-6799-41d7-aa67-6e7248112c90">

Does it help if you increase the spacing between the two rows, or even if you switch the order of the rows so that the interval option is at the bottom?